### ISSUE TITLE
Docs: broken link to rule list in quickstart

### ISSUE THREAD

**Issue by sandersn** [typescript-eslint/typescript-eslint#8480]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

https://typescript-eslint.io/getting-started/#details links to https://typescript-eslint.io/getting-started/Rules but this page doesn't exist.

### Affected URL(s)

https://typescript-eslint.io/getting-started/

**Comment by sandersn** [comment]
Going back and reloading the page, this now links to https://typescript-eslint.io/rules/, and works. What is going on?

**Comment by auvred** [comment]
Yeah, this link is actually broken, it should point on `/rules`, not on `/getting-started/Rules`

![broken rules link](https://github.com/typescript-eslint/typescript-eslint/assets/61150013/4e0e5322-6773-48b1-8f6e-3f1ba781d64b)

Perhaps we can fix this in https://github.com/typescript-eslint/typescript-eslint/pull/8468, as it fixes the broken link on the sibling line

**Comment by bradzacher** [comment]
just published the fix to the website
